faust/transport/consumer.py

```diff
@@ -1097,6 +1097,18 @@ def _new_offset(self, tp: TP) -> Optional[int]:
                 acked.extend(gaps)
                 gap_for_tp[:gap_index] = []
             acked.sort()
+
+            # We iterate over it until we handle gap in the head of acked queue
+            # then return the previous committed offset.
+            # For example if acked[tp] is:
+            #    34 35 36 37
+            #  ^-- gap
+            # self._committed_offset[tp] is 31
+            # the return value will be: 31
+            if self._committed_offset[tp]:
+                if min(acked) - self._committed_offset[tp] > 0:
```

Comment:
What if there is gap for that offset in the kafka topic? Do we have to check for that?

Reply:
I think that should be handled already since the gaps are combined into acked above. @s3563602123 can you add a test case to confirm?

Reply:
Added test cases to verify this situation